osquery/carver/carver.cpp

```diff
@@ -265,7 +265,9 @@ Status Carver::postCarve(const boost::filesystem::path& path) {
   // Construct the uri we post our data back to:
   auto startUri = TLSRequestHelper::makeURI(FLAGS_carver_start_endpoint);
   Request<TLSTransport, JSONSerializer> startRequest(startUri);
-  startRequest.setOption("hostname", FLAGS_tls_hostname);
+  if (FLAGS_carver_start_endpoint.rfind("/", 0) == 0) {
```

Comment:
What happens if someone sets the endpoint to `carver` ?

Reply:
I've inverted the logic to test if the start endpoint begins with `"http://"` or `"https://"`. I would like to update the appropriate documentation for the flag, but the best documentation I've found so far has been an issue that says we should add documentation (https://github.com/osquery/osquery/issues/3797). Do you have any suggestions on that front?